### PR TITLE
[NCL-5030] Possible to create invalid repository configurations

### PR DESCRIPTION
**Pull request by jomrazek**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [x] Have you added unit tests for your change?

### CODE REVIEW

**Comment by jomrazek** [comment]
retest this please

File: facade/src/main/java/org/jboss/pnc/facade/providers/SCMRepositoryProviderImpl.java [hunk 1]
```diff
@@ -154,6 +154,8 @@ public RepositoryCreationResponse createSCMRepository(String scmUrl, Boolean pre
         log.trace("Received request to start RC creation with url autodetect: " + scmUrl + " (sync enabled? " + preBuildSyncEnabled + ")");
         if(StringUtils.isEmpty(scmUrl)) throw new InvalidEntityException("You must specify the SCM URL.");
 
+        if(scmUrl.contains(config.getInternalScmAuthority()) && scmUrl.contains("/gerrit/")) throw new InvalidEntityException("Incorrect format of internal repository.");
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/providers/SCMRepositoryProviderImpl.java]: move to validateInternalRepository

**Comment by jomrazek** [comment]
test this

**Comment by jomrazek** [comment]
retest this please

**Comment by jomrazek** [comment]
retest this please

**Pull request merged**